### PR TITLE
CAP Plugin: cap-js ord plugin

### PR DESCRIPTION
**Pull request by RoshniNaveenaS**
This PR has documentation for the new cap ord plugin.

### CODE REVIEW

**Comment by renejeglinsky** [comment]
Hi @RoshniNaveenaS ,

could you please provide an mp4 instead of a gif? We have a job that optimizes the mp4 then.

**Comment by renejeglinsky** [comment]
Is this a plugin like all the others on this page? I'm missing the package name and I'm also not aware that a plugin can provide an enhancement to `cds compile`. Could you clarify?

**Comment by RoshniNaveenaS** [comment]
> Is this a plugin like all the others on this page? I'm missing the package name and I'm also not aware that a plugin can provide an enhancement to `cds compile`. Could you clarify?

I will check again with @chgeo for the same

**Comment by chgeo** [comment]
> Is this a plugin like all the others on this page? I'm missing the package name and I'm also not aware that a plugin can provide an enhancement to `cds compile`. Could you clarify?

Yes.  [This is how the ORD plugin does it](https://github.com/cap-js/ord/blob/bfae5156d401c1ffda8519fa1273e40d040b540b/cds-plugin.js).

File: plugins/index.md [hunk 1]
```diff
@@ -335,4 +335,42 @@ Available for:
...
+const csn = await cds.load(cds.env.folders.srv)
+const ord = cds.compile.to.ord(csn)
+```
+
+#### Command Line
+
+```sh
+cds compile <path to srv folder> --to ord [-o] [destinationFilePath]
+```
+<img width="1300" alt="Sample Application Demo" style="border-radius:0.5rem;" src="./assets/index/ordCLI.png">
```
Comment (renejeglinsky) [inline plugins/index.md:363]: I think we don't need this here, it's sufficient in the plugin's readme.

File: plugins/index.md [hunk 2]
```diff
@@ -335,4 +335,42 @@ Available for:
 
 <div id="planned-plugins" />
 
+## ORD (Open Resource Discovery)
+
+Open Resource Discovery (ORD) is a protocol that allows applications and services to self-describe their exposed resources and capabilities.It can be used to describe static documentation, but can also reflect tenant specific configuration and extensions (at run-time).​
```
Comment (renejeglinsky) [inline plugins/index.md:340]:
```suggestion
Open Resource Discovery (ORD) is a protocol that enables applications and services to self-describe their available resources and capabilities. It's not only useful for describing static documentation, but it also accurately reflects tenant-specific configurations and extensions at runtime.
```

File: plugins/index.md [hunk 3]
```diff
@@ -335,4 +335,42 @@ Available for:
 
 <div id="planned-plugins" />
 
+## ORD (Open Resource Discovery)
+
+Open Resource Discovery (ORD) is a protocol that allows applications and services to self-describe their exposed resources and capabilities.It can be used to describe static documentation, but can also reflect tenant specific configuration and extensions (at run-time).​ 
+Typically, ORD is used to describe APIs and Events, but it also supports higher-level concepts like Entity Types (Business Objects) and Data Products (beta).
+By adopting ORD, an application will implement a single-entry point (Service Provider Interface) that can be used to discover and crawl the relevant information / metadata. The information can be used to build a static metadata catalog or do detailed runtime inspection of actual system landscapes.
```
Comment (renejeglinsky) [inline plugins/index.md:342]:
```suggestion
When you adopt ORD, your application gains a single entry point, known as the Service Provider Interface. This interface allows you to discover and gather relevant information or metadata. You can use this information to construct a static metadata catalog or to perform a detailed runtime inspection of your actual system landscapes.
```

File: plugins/index.md [hunk 4]
```diff
@@ -335,4 +335,42 @@ Available for:
...
+<img width="1300" alt="Sample Application Demo" style="border-radius:0.5rem;" src="./assets/index/ordCLI.png">
+
+#### ORD Endpoints
+
+1) Run `cds watch` in the application's root.
+2) Check the following relative paths for ORD information - `/.well-known/open-resource-discovery` , `/open-resource-discovery/v1/documents/1`.
+
+<img width="1300" alt="Sample Application Demo" style="border-radius:0.5rem;" src="./assets/index/ordEndpoint.gif">
+
+
```
Comment (renejeglinsky) [inline plugins/index.md:372]: Add the log and link to the package as for any other package on this page.

File: plugins/index.md [hunk 5]
```diff
@@ -335,4 +335,42 @@ Available for:
...
+```sh
+cds compile <path to srv folder> --to ord [-o] [destinationFilePath]
+```
+<img width="1300" alt="Sample Application Demo" style="border-radius:0.5rem;" src="./assets/index/ordCLI.png">
+
+#### ORD Endpoints
+
+1) Run `cds watch` in the application's root.
+2) Check the following relative paths for ORD information - `/.well-known/open-resource-discovery` , `/open-resource-discovery/v1/documents/1`.
```
Comment (renejeglinsky) [inline plugins/index.md:368]: Just to check if I understand this:
I have to check a URL (which I have to know) and copy a URL out of the response. Then I open the copied URL. From there I copy a URL and open that one and that is then what I want to see. Correct?
If yes, can that be easier?

**Review by renejeglinsky — COMMENTED** [review]
Please move this document before all the `<div>` elements. Thank you.

**Comment by muskansethi1** [comment]
Hi @RoshniNaveenaS,
As far as I recall, we were putting this in internal documentation. Is there any change in the decision?

**Comment by RoshniNaveenaS** [comment]
> Hi @RoshniNaveenaS, As far as I recall, we were putting this in internal documentation. Is there any change in the decision?

I have set-up a call with @renejeglinsky and @chgeo for the same.
Will have a look at the review comments and make changes accordingly in the updated place

File: plugins/index.md (hunk 5 above)
Comment (RoshniNaveenaS) [reply plugins/index.md:368]: As discussed, this is the current functionality planned for the first release of ord plugin

File: plugins/index.md (hunk 1 above)
Comment (RoshniNaveenaS) [reply plugins/index.md]: have removed and linked to plugin readme

**Comment by renejeglinsky** [comment]
Hi,

I moved some content to the readme, see linked PR. Also I added the screenshot and moved the plugin further down. The first plugins mentioned on this page are forming some kind of special group, so even if plugins would be related, I'd not put them here.
Let me know what you think.

**Comment by RoshniNaveenaS** [comment]
> added the screenshot and moved the plugin further down. The first plugins mentioned on this page are forming some kind of special group, so even if plugins would be related, I'd not put them here.

Hello @renejeglinsky ,
Yeah this works.
I have merged the PR in ORD repo as well.
Let me know if this PR looks good

**Review by renejeglinsky — APPROVED** [review]

**Pull request merged**